### DISCUSSION TITLE
V5 auto-uninstall old version when UpgradeCode is the same

### DISCUSSION THREAD

**Discussion by fxtech** [wixtoolset/issues#8099] [Questions] [answered]
I'm using the new \<Files\> element which works well. I have an UpgradeCode. But when I install my .msi more than once, each time a new entry in the ARP (Add/Remove Programs) is created, and I have to remove them all independently, even if the Version string doesn't change.
Some documentation I found seems to indicate that the old stuff should be removed automatically when "upgrading", but thart doesn't seem to be the case.
Is there an example that shows how to properly manage this,so that only one copy of the program shows up in ARM, no matter how many times I "install" it?

**Comment by barnson** [comment, marked as answer]
Increment the version number and upgrades will work as you expect. Or take a look at [MajorUpgrade/@AllowSameVersionUpgrades](https://wixtoolset.org/docs/schema/wxs/majorupgrade/).

**Comment by fxtech** [comment]
Aha, thanks for that pointer to AllowSameVersionUpgrades!

Curious - why isn't that on by default? What is the point in allowing the same installer to be 'installed' more than once?

**Reply by robmen** [reply]
I dedicated two [Deployment Dojo](https://robmensching.com/deployment-dojo/) episodes to this topic. See [S1:E5](https://robmensching.com/deployment-dojo/episodes/s1/e5/the-problem-with-same-version-upgrades/) then [S1:E6](https://robmensching.com/deployment-dojo/episodes/s1/e6/the-real-problem-with-same-version-upgrades-in-wix-v4-copy/).

**Reply by fxtech** [reply]
Thanks Rob - very informative!

**Comment by robmen** [comment]
I think we're going to leave conversations open since they show up better in search. Marking the "Answer" is much appreciated though. Thanks.
